pkg/route/registry/route/rest.go

```diff
@@ -78,6 +79,13 @@ func (rs *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, er
 		route.Name = uuid.NewUUID().String()
 	}
 
+	//  shards will be eventually allocated via a separate controller.
+	shard := allocator.Allocate(route)
+
+	if len(route.Host) == 0 {
```

Comment:
I think this should occur prior to `ValidateRoute`.  That way we can ensure that whatever is generated passes the same validation as a user submitted name (`util.IsDNSSubdomain`).